middleware/jwt.go

```diff
@@ -25,12 +26,44 @@ import (
 // From https://github.com/dgrijalva/jwt-go/pull/139 it is understood
 // that if the machine where jwt token is generated and/or the machine
 // where jwt token is verified have any clock skew then there is a
-// possibility of getting a "Token used before issued" error. Considering
-// that we also check for expiration with delta of 5 minutes, disabling
-// iat claim until the patch is merged in jwt.
-var (
-	required_claims = []string{"iss", "exp"}
-)
+// possibility of getting a "Token used before issued" error.
+// Therefore we have implemented a derived claim type of HeketiJwtClaims and
+// the validation function for the same skips checking for "iat".
+// Also, now that we control the validation function, we make "exp" validation
+// mandatory and get rid of required_claims array.
+type HeketiJwtClaims struct {
+	*jwt.StandardClaims
+	Qsh string `json:"qsh,omitempty"`
+}
+
+func (c *HeketiJwtClaims) Valid() error {
+	vErr := new(jwt.ValidationError)
+	now := jwt.TimeFunc().Unix()
+
+	if c.VerifyExpiresAt(now, true) == false {
+		delta := time.Unix(now, 0).Sub(time.Unix(c.ExpiresAt, 0))
+		vErr.Inner = fmt.Errorf("Token is expired by %v", delta)
+		vErr.Errors |= jwt.ValidationErrorExpired
+	}
+
+	// We skip the check for "iat"
```

Comment:
We should either eliminate, modify, or conditionalize this block but we should not leave it just as a comment.